server/publications/teams.js

```diff
@@ -2,7 +2,19 @@ import { Meteor } from 'meteor/meteor';
 import { isAdmin, isVolunteer } from '../../lib/imports/method-helpers.js';
 
 Meteor.publish('admin.teams', function() {
-  return isAdmin(this.userId) ? Teams.find({}) : this.ready();
+  if (!isAdmin(this.userId)) return this.ready();
+
+  const projection = {
+    name: 1,
+    division: 1,
+    checkinConfirmed: 1,
+    inPerson: 1,
+    hasBegun: 1,
+    'puzzles.start': 1,
+    'puzzles.end': 1,
+    'puzzles.name': 1,
+  }
+  return Teams.find({}, {fields: projection});
```

Comment:
What is the purpose of the projection?

Reply:
This is similar to a projection in SQL (I think I stole their terminology, actually - mongo may call it something else). Basically, it's just "only return these fields." The problem with the team page being super slow last year, I think, was that we were serving (and monitoring changes for) the ENTIRETY of EVERY team in the game. This projection says "only return these specific fields." This saves us sending data, but as I understand it this also means that Meteor will have to react to fewer changes, so we'll have less frequent updates pushed out if we view the page. Specifically, the server will only push an update if one of the fields in "projection" changes, instead of if any part of any team changes. We'll see what this actually does for us in practice...

Reply:
Oh, awesome! I hope that'll help performance quite a bit this year.